checker/src/org/checkerframework/checker/index/upperbound/UpperBoundVisitor.java

```diff
@@ -49,6 +52,29 @@ public Void visitArrayAccess(ArrayAccessTree tree, Void type) {
         return super.visitArrayAccess(tree, type);
     }
 
+    /** Warns about LTLengthOf annotations with arguments whose lengths do not match. */
+    @Override
+    public Void visitAnnotation(AnnotationTree node, Void p) {
+        AnnotationMirror anno = TreeUtils.annotationFromAnnotationTree(node);
+        if (AnnotationUtils.areSameByClass(anno, LTLengthOf.class)) {
+            List<? extends ExpressionTree> args = node.getArguments();
+            if (args.size() == 2) {
+                // if offsets are provided, there must be the same number of them as there are arrays
+                List<String> sequences =
+                        AnnotationUtils.getElementValueArray(anno, "value", String.class, true);
+                List<String> offsets =
+                        AnnotationUtils.getElementValueArray(anno, "offset", String.class, true);
+                if (sequences.size() != offsets.size() && offsets.size() > 0) {
```

Comment:
Why allow more offsets than sequences?  Also, this requirement should be documented in `LTLengthOf.java`.

Reply:
This doesn't allow more offsets than sequences. An error is issued when the number of offsets isn't equal to the number of sequences and the number of offsets is non-zero (the second condition is necessary to allow `@LTLengthOf("a")`, which has zero offsets but a non-zero number of sequences). I'll add a test to cover that condition, though.

Reply:
Oh right.  The error key is "not.enough.offsets", which is confusing because it will be issued when there are too many offsets, too.

Reply:
Good point. I'll change it to "wrong.number.of.offsets" or something like that instead.

Reply:
`offset.not.equal.sequence`, maybe?

Reply:
What about `different.length.sequences.offsets`? I think I like that better and it's more clear about what the problem is.

Reply:
Sounds good.